### PR TITLE
QTY-1300 Make the shim work without shim patch locally

### PR DESCRIPTION
**Pull request by shuynh3**
[QTY-1300](https://strongmind.atlassian.net/browse/QTY-1300)

## Purpose 
Removes the need to apply the shim patch to get canvas running locally

## Approach 
Applies the changes the shim does dynamically based on environment

## Testing
Integration testing, tested common functionality and JQuery

### CODE REVIEW

**Review by adiaz04 — APPROVED** [review]

File: app/services/settings_service/repository_base.rb [hunk 1]
```diff
@@ -30,7 +30,11 @@ def create_table(name:)
     end
 
     def dynamodb
-      @dynamodb || Aws::DynamoDB::Client.new
+      if ENV['RAILS_ENV'] == 'development'
```
Comment (CJAdeszko) [inline app/services/settings_service/repository_base.rb:33]: I'm wondering if there is a cleaner way to write this? Something like
```
@dynamodb ||= Rails.env.development? ? Aws::DynamoDB::Client.new(:endpoint => "http://dynamodb:8000/") : Aws::DynamoDB::Client.new
```

I don't know if that will actually work or read better in practice so I could be swayed either way. Just a thought so we don't have multiple `@dynamodb ||=` statements...

Comment (CJAdeszko) [reply app/services/settings_service/repository_base.rb:33]: Curious if @adiaz04 has an opinion...

Comment (adiaz04) [reply app/services/settings_service/repository_base.rb:33]: I like it. 👍

**Review by CJAdeszko — APPROVED** [review]
Looks good to me 👍

**Comment by CJAdeszko** [comment]
I approved the PR, although, I think we need to add a description to the PR itself @shuynh3

**Pull request merged**